ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/PeerDiscoveryAgent.java

```diff
@@ -125,25 +124,14 @@ public PeerDiscoveryAgent(
       LOG.info("Starting peer discovery agent on host={}, port={}", host, port);
 
       // override advertised host if we detect an external IP address via NAT manager
-      String externalIpAddress = null;
-      if (natManager.isPresent()) {
-        try {
-          final int timeoutSeconds = 60;
-          LOG.info("Waiting for up to {} seconds to detect external IP address...", timeoutSeconds);
-          externalIpAddress =
-              natManager.get().queryExternalIPAddress().get(timeoutSeconds, TimeUnit.SECONDS);
-
-        } catch (Exception e) {
-          LOG.warn(
-              "Caught exception while trying to query NAT external IP address (ignoring): {}", e);
-        }
+      final String advertisedAddress;
+      if (natService.isPresent()) {
+        advertisedAddress =
+            natService.get().queryExternalIPAddress().orElseGet(config::getAdvertisedHost);
+      } else {
+        advertisedAddress = config.getAdvertisedHost();
```

Comment:
Now that the logic is simpler If you use a `flatMap` on `natService` here you don't need to declare and assign separately

Reply:
Done